pycbc/fft/backend_cupy.py

```diff
@@ -0,0 +1,37 @@
+#  Copyright (C) 2014 Josh Willis
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with with program; see the file COPYING. If not, write to the
+#  Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#  MA  02111-1307  USA
+
+from .core import _list_available
+
+_backend_dict = {'cupy' : 'cupyfft'}
+_backend_list = ['cupy']
+
+_alist, _adict = _list_available(_backend_list, _backend_dict)
```

Comment:
The `backend_cuda` version of this has the `if pycbc.HAVE_CUDA` statement and this doesn't. This makes me think, should this backend work when _not_ on a GPU?

Reply:
I think that's used to stop the tests failing here when no GPU is present by not loading any CUDA module ... I'll probably need this, but it *might* be possible to have the documentation and help text run for the cupy backend, even if the code isn't going to work.

Reply:
(Clearly the tests do need to pass before merging).